server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -180,16 +201,23 @@ public TransportShardBulkAction(
         this.mappingUpdatedAction = mappingUpdatedAction;
         this.segmentReplicationPressureService = segmentReplicationPressureService;
 
-        this.transportPrimaryTermValidationAction = ACTION_NAME + "[validate_primary_term]";
-
-        transportService.registerRequestHandler(
-            transportPrimaryTermValidationAction,
-            executor,
-            true,
-            true,
-            PrimaryTermValidationRequest::new,
-            this::handlePrimaryTermValidationRequest
-        );
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)) {
+            this.latestSuccessfulLeaderCheckerTimeSupplier = () -> ((Coordinator) discovery).getLatestSuccessfulLeaderCheckerTime();
```

Comment:
The setting value should be validated against follower check interval

Reply:
Cluster manager will acknowledge a leader checker from a node which it has in it's discovery nodes (which it only allows to add when a join request succeeds). But curious on knowing your take on validation against the follow checker.